ci: fix and harden SonarCloud workflow

oxr463 opened the pull request:
Fix the SonarCloud CI workflow and bring it up to current standards.
  
  - Fix broken action reference: `sonarsource/sonarcloud-github-c-cpp` was
    renamed to `SonarSource/sonarqube-github-c-cpp`; update to `@v2`
  - Fix SonarCloud scanner configuration
  - Rename `main.yml` → `pull-request.yml` to reflect the workflow's purpose
  - Pin all external actions to full commit SHAs (S7637)

EIGHT880 commented on .github/workflows/main.yml:
```diff
@@ -53,6 +52,8 @@ jobs:
       - name: Run sonar-scanner
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
```
- name: Run sonar-scanner
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        run: |
          find /home/runner/work/proot/proot/test/ -type d -exec chmod 755 {} \;
          find /home/runner/work/proot/proot/test/ -type f -exec chmod 644 {} \;
          # ... rest of your build-wrapper command

oxr463 committed:
ci(sonarcloud): fix sonarcloud scanner configuration

- set sonar.sources and sonar.host.url in sonar-project.properties
- update build command to use clean build without separate coverage flags
- fix secret name from SONARCLOUD_TOKEN to SONAR_TOKEN
- add chmod steps for test directory before sonar-scanner run

oxr463 committed:
ci: fix sonarqube c/cpp action rename

oxr463 committed:
ci: rename main.yml to pull-request.yml

oxr463 committed:
ci: pin actions to full commit SHA hashes
